draft-ietf-quic-http.md

```diff
@@ -97,38 +97,82 @@ defined in {{!RFC2119}}.
+the following header:
+
+    Alt-Svc: hq=":443"; v="Q034,Q033"
+
+Where multiple versions are listed, the order of the values reflects the
+server's preference (with the first value being the most preferred version).
+
+QUIC versions are four-octet sequences with no additional constraints on format. 
+Octets not allowed in tokens ({{!RFC7230}}, Section 3.2.6) MUST be 
+percent-encoded as per Section 2.1 of {{!RFC3986}}. Consequently, the octet 
+representing the percent character ("%", hex 25) MUST be percent-encoded as 
+well. 
+
+On receipt of this header, a client MAY attempt to establish a QUIC connection
 on port 443 and, if successful, send HTTP requests using the mapping described
-in this document.
+in this document.  Servers SHOULD list only versions which they support, but MAY
+omit supported versions for any reason.
 
 Connectivity problems (e.g. firewall blocking UDP) may result in QUIC connection
 establishment failure, in which case the client should gracefully fall back to
 HTTP/2.
 
+# Connection establishment {#connection-establishment}
 
-# Connection establishment
-
-HTTP/QUIC connections are established as described in {{QUIC-TRANSPORT}}. The 
-QUIC crypto handshake MUST use TLS {{QUIC-TLS}}. 
+HTTP/QUIC connections are established as described in {{QUIC-TRANSPORT}}. During 
+connection establishment, HTTP/QUIC support is indicated by selecting the ALPN 
+token "hq" in the crypto handshake. 
 
 While connection-level options pertaining to the core QUIC protocol are set in 
-the initial crypto handshake {{QUIC-TLS}}, HTTP-specific settings are conveyed 
+the initial crypto handshake, HTTP-specific settings are conveyed 
 in the SETTINGS frame. After the QUIC connection is established, a SETTINGS 
 frame ({{frame-settings}}) MUST be sent as the initial frame of the HTTP control
 stream (StreamID 3, see {{stream-mapping}}).
 
+## Draft version identification
+   
+> **RFC Editor's Note:**  Please remove this section prior to publication of a
+> final version of this document.
+
+Only implementations of the final, published RFC can identify themselves as 
+"hq". Until such an RFC exists, implementations MUST NOT identify themselves 
+using these strings. 
+
+Implementations of draft versions of the protocol MUST add the string "-" and 
+the corresponding draft number to the identifier. For example, 
+draft-ietf-quic-http-01 is identified using the string "hq-01". 
+
+Non-compatible experiments that are based on these draft versions MUST append 
+the string "-" and an experiment name to the identifier. For example, an 
+experimental implementation based on draft-ietf-quic-http-09 which reserves an 
+extra stream for unsolicited transmission of 1980s pop music might identify
+itself as "hq-09-rickroll". Note that any label MUST conform to the "token"
+syntax defined in Section 3.2.6 of [RFC7230]. Experimenters are encouraged to
+coordinate their experiments on the ietf-quic-wg@w3.org mailing list.
```

Comment:
Should the mailing list be `quic@ietf.org`? The one listed above doesn't seem to exist https://lists.w3.org/Archives/Public/ietf-quic-wg/